proto/desmos/subspaces/v1beta1/msgs.proto

```diff
@@ -0,0 +1,206 @@
+syntax = "proto3";
+package desmos.subspaces.v1beta1;
+
+import "gogoproto/gogo.proto";
+
+option go_package = "github.com/desmos-labs/desmos/x/staging/subspaces/types";
+
+// Msg defines subspaces Msg service.
+service Msg {
+
+  // CreateSubspace defines the method to create a subspace
+  rpc CreateSubspace(MsgCreateSubspace) returns (MsgCreateSubspaceResponse);
+
+  // EditSubspace defines the method to edit a subspace fields
+  rpc EditSubspace(MsgEditSubspace) returns (MsgEditSubspaceResponse);
+
+  // AddAdmin defines the method to add a new admin to the subspace
+  rpc AddAdmin(MsgAddAdmin) returns (MsgAddAdminResponse);
+
+  // RemoveAdmin defines the method to remove an admin from a specific subspace
+  rpc RemoveAdmin(MsgRemoveAdmin) returns (MsgRemoveAdminResponse);
+
+  // RegisterUser defines the method to let user posts inside a specific subspace
+  rpc RegisterUser(MsgRegisterUser) returns (MsgRegisterUserResponse);
+
+  // UnregisterUser defines the method to unregister a user from a subspace
+  rpc UnregisterUser(MsgUnregisterUser) returns (MsgUnregisterUserResponse);
+
+  // BanUser defines the method to ban a user inside a specific subspace
+  rpc BanUser(MsgBanUser) returns (MsgBanUserResponse);
+
+  // UnbanUser defines the method to unban a user inside a specific subspace
+  rpc UnbanUser(MsgUnbanUser) returns (MsgUnbanUserResponse);
+
+}
+
+// ___________________________________________________________________________________________________________________
+
+// MsgCreateSubspace represents the message used to create a subspace
+message MsgCreateSubspace {
+  option (gogoproto.equal) = false;
+  option (gogoproto.goproto_getters) = false;
+
+  string id = 1 [
+    (gogoproto.customname) = "SubspaceID",
+    (gogoproto.jsontag) = "subspace_id",
+    (gogoproto.moretags) = "yaml:\"subspace_id\""
+  ];
+
+  string name = 2 [
+    (gogoproto.moretags) = "yaml:\"name\""
+  ];
+
+  bool open = 3 [ (gogoproto.moretags) = "yaml:\"open\""];
+
+  string creator = 4 [ (gogoproto.moretags) = "yaml:\"creator\"" ];
+}
+
+// MsgCreateSubspaceResponse defines the Msg/CreateSubspace response type
+message MsgCreateSubspaceResponse {}
+
+// ___________________________________________________________________________________________________________________
+
+// MsgEditSubspace represents the message used to edit a subspace fields
+message MsgEditSubspace {
```

Comment:
I was thinking the same and I stopped while thinking about how to handle a default value of it, if the user doesn't specify it while making the tx.
Suppose we have an open subspace and we want to edit only its owner.   
What if the default value of the `--open` flag is `false`?
It will edit the open field anyway even though the user doesn't want that.
Do you think we can use a flag type `string` and converting it later inside the message itself to solve this?

Reply:
@bragaz You're right, I didn't think about this. What I think we can do to solve this is, instead of using an `open` field, we can use a `type` field. We can then have `SUBSPACE_TYPE_OPEN` and `SUBSPACE_TYPE_CLOSED` to identify the various types of subspaces. These should be implemented as an `enum` similar to [the `x/gov` one inside the Cosmos SDK](https://github.com/cosmos/cosmos-sdk/blob/master/proto/cosmos/gov/v1beta1/gov.proto#L17)